# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../dietlibc/strnlen.patch
# Copyright (C) 2004 - 2005 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---

Needed by latest u/dev ...

  - Rene Rebe <rene@exactcode.de>

--- /dev/null	2005-11-24 11:09:35.213013250 +0100
+++ dietlibc-0.29/lib/strnlen.c	2005-11-26 23:48:26.000000000 +0100
@@ -0,0 +1,10 @@
+#include <endian.h>
+#include "dietfeatures.h"
+#include <string.h>
+
+size_t strnlen(const char *s, size_t maxlen) {
+  register size_t i;
+  if (__unlikely(!s)) return 0;
+  for (i=0; __likely(i < maxlen) && __likely(*s); ++s) ++i;
+  return i;
+}
